include/nbl/video/IGPUDescriptorSet.h

```diff
@@ -112,6 +140,20 @@ class IGPUDescriptorSet : public asset::IDescriptorSet<const IGPUDescriptorSetLa
             return samplers + localOffset;
         }
 
+        // Same as above, but amortizes lookup if you already have an index
+        inline core::smart_refctd_ptr<IGPUSampler>* getMutableCombinedSamplersIndexed(const uint32_t bindingStorageIndex) const
+        {
+            const auto localOffset = getLayout()->getMutableCombinedSamplerRedirect().getStorageOffset(redirect_t::storage_range_index_t{ bindingStorageIndex }).data;
+            if (localOffset == getLayout()->getMutableCombinedSamplerRedirect().Invalid)
+                return nullptr;
+
+            auto* samplers = getAllMutableCombinedSamplers();
+            if (!samplers)
+                return nullptr;
+
+            return samplers + localOffset;
+        }
```

Comment:
comments same as for https://github.com/Devsh-Graphics-Programming/Nabla/pull/693/commits/8aef486c514ea26f35361afcf74fa153c1320da0#r1659354882